rpc_jobs/bump_snapshots.yml

```diff
@@ -0,0 +1,146 @@
+        dir("rpc-openstack"){
+          git branch: 'master', url: 'https://github.com/rcbops/rpc-openstack'
+
+          for ( release in releases ) {
+
+            String latest = sh(
+              returnStdout: true,
+              script: """#!/bin/bash -xe
+                git tag --list | egrep "r${release['major']}\\." | sort --version-sort -r | head -1
+              """
+            ).trim()
+
+            String previous = sh(
+              returnStdout: true,
+              script: """#!/bin/bash -xe
+                git tag --list | egrep "r${release['major']}\\." | sort --version-sort -r | head -2 | tail -1
+              """
+            ).trim()
+
+            release['latest'] = latest
+            release['previous'] = previous
+          } // for
+        } // dir
+        dir("rpc-gating"){
+          git branch: env.RPC_GATING_BRANCH, url: env.RPC_GATING_REPO
+
+          def parallelBuilds = [:]
+
+          for ( r in releases ) {
+            def release = r
+            println "series => ${release['series']}, latest => ${release['latest']}, pervious => ${release['previous']}"
+
+            parallelBuilds["${release['series']}-${release['image']}"] = {
+              build(
+                job: 'PM_rpc-gating-master-snapshot-snapshot-test',
+                wait: true,
+                parameters: [
+                  [
+                    $class: "StringParameterValue",
+                    name: "RPC_GATING_BRANCH",
+                    value: env.RPC_GATING_BRANCH,
+                  ],
+                  [
+                    $class: "StringParameterValue",
+                    name: "IMAGE",
+                    value: "rpc-${release['latest']}-${release['image']}",
+                  ]
+                ]
+              )
+            } // parallelBuilds
+          } // for
+
+          parallel parallelBuilds
+
+          for ( release in releases ) {
+            sh """#!/bin/bash -xe
+              sed -i "s/rpc-${release['previous']}-${release['image']}/rpc-${release['latest']}-${release['image']}/g" rpc_jobs/*.yml
+            """
+          } // for
+
+          withCredentials(
+            [
+              string(
+                credentialsId: 'rpc-jenkins-svc-github-pat',
+                variable: 'PAT'
+              ),
+              usernamePassword(
+                credentialsId: "jira_user_pass",
+                usernameVariable: "JIRA_USER",
+                passwordVariable: "JIRA_PASS"
+              ),
+            ]
+          ) {
+            sshagent (credentials:['rpc-jenkins-svc-github-ssh-key']) {
+              // NOTE(matt): We end up with untracked files in this repo, so
+              // we skip showing untracked files when we run git status
+              String git_status = sh(
+                returnStdout: true,
+                script: """#!/bin/bash -xe
+                  git status -s -uno
+                """
+              ).trim()
+
+              if (git_status) {
```

Comment:
You might be able to replace most of this section with scripts/commit_and_pull_request.sh.

Reply:
Updated to use that instead.